Content/teams debugger docs (#842)
trufflesuite/trufflesuite.com: src/docs/teams/debugger/debugging-a-transaction.md

---
title: Teams | Debugging a Transaction
layout: docs.hbs
---
# Debugging a Transaction

<p class="alert alert-info">

> <strong>Note</strong>: Truffle Teams currently only allows you to debug transactions on <a href="/docs/teams/sandboxes/sandboxes-overview">Ganache sandboxes</a>. Support for public networks (such as the <code>Görli testnet</code> or <code>mainnet</code>) will be available soon.

Reviewer: ```suggestion
<strong>Note</strong>: Truffle Teams currently only allows you to debug transactions on <a href="/docs/teams/sandboxes/sandboxes-overview">Ganache sandboxes</a>. Support for public networks (such as the <code>Görli</code> testnet or <code>mainnet</code>) will be available soon.
```

Author: i'll go ahead and commit this (it was actually initially how I'd highlighted it), although it kinds looks slightly off given we're highlighting `mainnet`...minor thing though 🤷‍♂️

Reviewer: It would be acceptable to not highlight either in this case

</p>

> The debugger is reached from the DEPLOYMENTS tab. From here, select the deployment which included the contracts that received the transaction that you are looking to debug. In the example screenshot below, this is `deployment-01`.

Reviewer: ```suggestion
The debugger is accessed from the DEPLOYMENTS tab. From here, [select the deployment](/docs/teams/deployments/deployment-details) which included the contracts that received the transaction that you are looking to debug. In the example screenshot below, this is `deployment-01`.
```

<figure class="screenshot">
  <img class="figure-shadow mb-2 w-100" src="/img/docs/teams/debugger-deployments.png" alt="Teams Deployments">
  <figcaption class="text-center">The Truffle Teams Deployments screen.</figcaption>
</figure>

> At this screen, you can either click **<span class="inline-button"><i class="fas fa-heart-rate"></i> MONITOR</span>**  to the right of the contract to view transactions sent solely to that contract address, or TRANSACTIONS to view all the transactions associated with the entire project.

Reviewer: ```suggestion
At this screen, you can either click **<span class="inline-button"><i class="fas fa-heart-rate"></i> MONITOR</span>**  to the right of the contract to view transactions sent solely to that contract address, or the **TRANSACTIONS** tab to view all the transactions associated with the entire deployment.
```

<figure class="screenshot">
  <img class="figure-shadow mb-2 w-100" src="/img/docs/teams/debugger-deployment.png" alt="Teams Deployment">
  <figcaption class="text-center">The Truffle Teams Deployment screen.</figcaption>
</figure>

Assuming your contract(s) have received one or more transactions you will now see a list similar to the screenshot below.

<figure class="screenshot">
  <img class="figure-shadow mb-2 w-100" src="/img/docs/teams/debugger-monitoring-transactions.png" alt="Teams Transaction Monitoring">
  <figcaption class="text-center">The Truffle Teams Transaction Monitoring screen.</figcaption>
</figure>

> The debugger can be started by clicking on the **<span class="inline-button"><i class="fas fa-debug"></i> DEBUG</span>** button located on the right-hand-side of a given transaction.

Reviewer: somewhere here would be a good spot to add that note about enabling early access if they don't see the button

Author: sure, i'll go ahead and add it

> Assuming your contract(s) have received one or more transactions you will now see a list similar to the screenshot below.
> 
> <figure class="screenshot">
>   <img class="figure-shadow mb-2 w-100" src="/img/docs/teams/debugger-monitoring-transactions.png" alt="Teams Transaction Monitoring">
>   <figcaption class="text-center">The Truffle Teams Transaction Monitoring screen.</figcaption>
> </figure>
> 
> The debugger can be started by clicking on the **<span class="inline-button"><i class="fas fa-debug"></i> DEBUG</span>** button located on the right-hand-side of a given transaction.

Reviewer: ```suggestion
Assuming your contract(s) have received one or more transactions you will now see a list similar to the screenshot below.

<figure class="screenshot">
  <img class="figure-shadow mb-2 w-100" src="/img/docs/teams/debugger-monitoring-transactions.png" alt="Teams Transaction Monitoring">
  <figcaption class="text-center">The Truffle Teams Transaction Monitoring screen.</figcaption>
</figure>

The debugger can be started by clicking on the **<span class="inline-button"><i class="fas fa-debug"></i> DEBUG</span>** button for the transaction you'd like to debug.
```

We try to refrain to use directional language. See this thread in Slack: https://trufflesuite.slack.com/archives/CG1DUUSJF/p1604532063170700

  <figcaption class="text-center">The Truffle Teams Transaction Monitoring screen.</figcaption>
</figure>

<p class="alert alert-info">
<strong>Note</strong>: The Truffle Teams Debugger is currently only available for transactions excuted sent to contracts in <a href="/docs/teams/sandboxes/sandboxes-overview">Ganache sandboxes</a> and accordingly the DEBUG button will show as disabled for other networks. Debugging transactions made against public networks will be available in a forthcoming release.
</p>

> 

Reviewer: you also need to add something about "if you don't see the button at all, you need to enable early access"

Author: 👌 done

Author: wrestled with the wording a little so tweaks totally welcome
